1029 - TYDE - Allow creating avatar and use it in app

odotadot opened the pull request:
## Changes
- Show an interface to the user asking them to customize their avatar the first time they use a tapestry
- Allow users to go in and edit their avatar at any point in the settings
- Update TYDE to use avatars where needed and create a simple way of viewing avatars where needed in the code
- Avatars are shared across all tapestries on a given site

## Screenshot
https://user-images.githubusercontent.com/34692466/123856151-9850d500-d8d5-11eb-8076-24c8d4a6b8bf.mp4
## Issue Linkage
Closes (issue #1029 )
## PR Dependency
(PR #1030 )
## Automated Testing

- Changing .env variable VUE_APP_AVATARS=TRUE will show the avatar customizer prompt upon entering tapestry if no avatar has been saved for user yet
- Can go to the User Settings (new toolbar button) to change Avatar and save it
- Saved Avatar look should reflect in the Avatar Customizer every time you open it

odotadot commented:
This branch is a copy of 'avataaars' but without the previous merge commits involving tyde2 that added many files under the 'Files changed' tab. It has all the relevant changes of the 'avataaars' branch.

wynnset commented:
@odotadot please rebase this branch on top of circle of support and change the one emoji that is at the center of the middle of the circle view to an avatar instead: 
![image](https://user-images.githubusercontent.com/8981258/128253959-01377355-0c56-4eb3-98c6-9a4347772e53.png)

wynnset commented on templates/vue/src/components/Toolbar/index.vue:
```diff
@@ -1,18 +1,27 @@
-      ></tapestry-depth-slider>
+    <tapestry-filter v-if="!showMap" style="z-index: 10" />
+    <div
+      v-show="isLoggedIn"
```
This erroneously hides the depth slider for non-logged in users.

odotadot replied:
Fixed

wynnset commented on templates/vue/src/components/Toolbar/index.vue:
```diff
@@ -1,18 +1,27 @@
+    <tapestry-filter v-if="!showMap" style="z-index: 10" />
+    <div
+      v-show="isLoggedIn"
+      :class="[{ 'hide-toolbar': hideToolbar }, 'slider-wrapper']"
```
Why would you hide via CSS when you can do the same using `v-show`?

odotadot replied:
Removed

wynnset commented on templates/vue/src/components/tyde/activities/CircleOfSupport/CircleView/index.vue:
```diff
@@ -73,6 +73,28 @@
+        :topType="avatar.topType"
+        :topColor="avatar.topColor"
+      ></avataaars>
+      <div v-else class="user">😊</div>
```
Can we use a default avatar instead of an emoji to keep consistent with the styling?

odotadot replied:
Replaced with default avatar

wynnset commented on templates/vue/src/components/tyde/activities/CircleOfSupport/CircleView/index.vue:
```diff
@@ -73,6 +73,28 @@
         src="https://twemoji.maxcdn.com/v/13.1.0/72x72/1f60a.png"
       />
     </div>
+    <div>
```
Not sure why this `<div>` wrapper is needed

odotadot replied:
removed

wynnset reviewed (changes requested):
This is looking great! I already merged in tyde2, so once you address these minor things, I can test it one more time and get it merged in. Thanks!

wynnset commented:
- [ ] This is not currently working with the TYDE mode on which needs to be fixed.

odotadot commented:
Addressed all comments. Avatar initialization should work for both TYDE and regular mode for all users. The only thing is that there isn't a button in TYDE mode for users to customize their avatar. Should this be added in? Otherwise, I think this PR looks good.

wynnset commented:
@EldadZZipori can you please fix the tests here? I am unable to get Cypress to work on my machine and therefore this is hard for me to do. Thanks!
